src/serialmonitor/usbDetector.ts

```diff
@@ -14,11 +14,22 @@ import { SerialMonitor } from "./serialMonitor";
 
 export class UsbDetector {
 
+    public static extensionRoot: string;
+
+    public static getInstance(): UsbDetector {
+        if (!UsbDetector._instance) {
+            UsbDetector._instance = new UsbDetector(UsbDetector.extensionRoot);

```

Comment:
I am not sure what behavior should be if the assumption is violated. Can you give me some advice?